comp/dogstatsd/server/parse.go

```diff
@@ -238,8 +246,25 @@ func (p *parser) parseFloat64List(rawFloats []byte) ([]float64, error) {
 }
 
 // extractContainerID parses the value of the container ID field.
+// If the field is prefixed by `in-`, it corresponds to the cgroup controller's inode of the source
+// and is used for ContainerID resolution.
 func (p *parser) extractContainerID(rawContainerIDField []byte) []byte {
-	return rawContainerIDField[len(containerIDFieldPrefix):]
+	containerIDField := rawContainerIDField[len(containerIDFieldPrefix):]
+
+	if bytes.HasPrefix(containerIDField[:len(containerIDFieldInodePrefix)], containerIDFieldInodePrefix) {
```

Comment:
Can we add some unit tests in `parse_test.go` that flex this codepath?

Reply:
Done in [77b8ad1](https://github.com/DataDog/datadog-agent/pull/21532/commits/77b8ad142863dae7ae01f506a61427cba687a8b6)

Reply:
Thanks, that test is useful! I was thinking more along the lines of testing the actual parsing code (and I gave you the wrong filename sorry 😓) something similar to this test https://github.com/DataDog/datadog-agent/blob/main/comp/dogstatsd/server/parse_metrics_test.go#L475-L492

Reply:
Done in https://github.com/DataDog/datadog-agent/pull/21532/commits/d1001447ab7c795670ad79baf9a9a388e87a30bc